### ISSUE TITLE
Using a drive instead of downloading files

### ISSUE THREAD

**Issue by brichet** [jupyterlite/litegitpuller#17]
Currently the files are downloaded to the file browser, and saved in the web browser storage (indexed DB).

After discussion with @martinRenou, we could probably use a custom drive instead of copying the files to the file browser.

Advantages / drawbacks to be discussed.

**Comment by jtpio** [comment]
> and saved in the web browser storage (indexed DB).

When used in JupyterLite, right? I suppose because by default JupyterLite saves files in IndexedDB.

This could also be relevant to https://github.com/jupyterlite/jupyterlite/issues/1041, if this behavior gets moved to a drive at some point too.

**Comment by martinRenou** [comment]
> When used in JupyterLite, right?

Right, if this extension was installed on a classic JupyterLab session, you would download the entire git repo onto the server

**Comment by westurner** [comment]
- https://github.com/jupyterlite/jupyterlite/issues/315
- https://github.com/jupyterlite/jupyterlite/issues/464
- https://github.com/jupyterlite/jupyterlite/pull/655
- https://github.com/jupyterlite/jupyterlite/pull/655/files#diff-2546ca49df64ebaaf92af83afaff210a89e6b51b157dfd0aa150d7e4d9643370
  - DriveFS, examples/pyolite/virtual_drive.ipynb
- https://github.com/jupyterlite/jupyterlite/issues/1041
- https://github.com/jupyterlite/jupyterlite/issues/430#issuecomment-1499359942
  - https://github.com/jupyterlab-contrib/jupyterlab-open-url-parameter
- https://github.com/jupyterlite/jupyterlite/tree/main/packages/contents *packages/contents*
- https://github.com/jupyterlite/jupyterlite/blob/main/packages/contents/src/drive.ts#L40 drive.ts  
  `export class BrowserStorageDrive implements Contents.IDrive {`
- https://github.com/jupyterlite/jupyterlite/blob/main/packages/contents/src/drivefs.ts drivefs.ts
- https://github.com/jupyterlab/jupyterlab-git
  - [ ] extract an interface to the git operations necessary for *jupyterlite* jupyterlab git
    - [ ] add tests for the [ isomorphic-git, jupyter-drives, jupyter-fs, IDrive implementer ] 
      - [ ] refactor existing tests as necessary
    - https://github.com/jupyterlab/jupyterlab-git/blob/5b5678bfdcd592d7220b869be68a56cecfa44e8b/jupyterlab_git/git.py#L1205 pull
    - https://github.com/jupyterlite/jupyterlite/pull/603/files
    - https://github.com/jupyterlab/jupyterlab-git/blob/main/jupyterlab_git/tests/test_clone.py 
    - https://github.com/jupyterlab/jupyterlab-git/blob/main/jupyterlab_git/tests/test_pushpull.py
    - https://github.com/jupyterlab/jupyterlab-git/blob/main/jupyterlab_git/tests/test_fetch.py
  - [ ] implement jupyterlab_git git.py with [ ... ]
- /? jupyterlab drive interface
  https://www.google.com/search?q=jupyterlab+drive+interface
  - ~https://github.com/jupyterlab/jupyterlab-google-drive~
  - https://github.com/QuantStack/jupyter-drives /blob/main/jupyter_drives/manager.py
    - [ ] jupyterlite?
  - https://github.com/jpmorganchase/jupyter-fs
    - > A plugin for JupyterLab that lets you set up and use as many filebrowsers as you like, connected to whatever local and/or remote filesystem-like resources you want.
      > 
      > The backend is built on top of PyFilesystem and fsspec, while the frontend is built on top of tree-finder.
    - [ ] jupyterlite?
  - https://jupyterlab-contrib.github.io/extensions.html
  - https://jupyterlab-contrib.github.io/jupyterlab-browser-storage.html https://github.com/jupyterlab-contrib/jupyterlab-browser-storage
  - https://jupyterlab-contrib.github.io/jupyterlab-filesystem-access.html https://github.com/jupyterlab-contrib/jupyterlab-filesystem-access
  - https://jupyterlab-contrib.github.io/jupyter-archive.html https://github.com/jupyterlab-contrib/jupyter-archive
  - https://jupyterlab-contrib.github.io/jupyterlab-fileopen.html https://github.com/jupyterlab-contrib/jupyterlab-fileopen
  - docs: 
    - JupyterLab API Docs > IDrive:
      https://jupyterlab.readthedocs.io/en/latest/api/interfaces/services.Contents.IDrive.html
